File: swift/Sources/ProjectNameSDK/entity/AccountUpdaterScheduleEntity.swift
// AccountUpdaterSchedule entity client for the BluefinPayconex SDK.

import Foundation

public final class AccountUpdaterScheduleEntity: BluefinPayconexEntityBase {
  public init(_ client: BluefinPayconexSDK, _ entopts: VMap? = nil) {
    super.init(client, entopts, "account_updater_schedule")
  }

  public override func make() -> Entity {
    return AccountUpdaterScheduleEntity(client, cloneOpts())
  }

  // (load not defined by this API - base class throws unsupportedOp)

  // (list not defined by this API - base class throws unsupportedOp)

  public override func create(_ reqdata: VMap?, _ ctrl: VMap?) throws -> Value {
    var ctxmap: [String: Any?] = ["opname": "create", "match": match, "data": data]
    if let ctrl = ctrl { ctxmap["ctrl"] = ctrl }
    if let reqdata = reqdata { ctxmap["reqdata"] = reqdata }
    let ctx = utility.makeContext(ctxmap, entctx)
  
    return try runOp(ctx) {
      if let result = ctx.result {
        if !isNil(result.resdata) {
          self.data = clone(result.resdata).asMap ?? VMap()
        }
      }
    }
  }

  // (update not defined by this API - base class throws unsupportedOp)

  // (remove not defined by this API - base class throws unsupportedOp)
}

File: csharp/entity/AccountUpdaterScheduleEntity.cs
// AccountUpdaterSchedule entity client for the BluefinPayconex SDK.

using Voxgig.Struct;

namespace BluefinPayconexSdk.Entity;

public class AccountUpdaterScheduleEntity : BluefinPayconexEntityBase
{
    public AccountUpdaterScheduleEntity(BluefinPayconexSDK client, Dictionary<string, object?>? entopts = null)
        : base(client, entopts, "account_updater_schedule")
    {
    }

    public override IEntity Make()
    {
        return new AccountUpdaterScheduleEntity(client, CloneOpts());
    }

    // (load not defined by this API - base class throws UnsupportedOp)

    // (list not defined by this API - base class throws UnsupportedOp)

    public override object? Create(Dictionary<string, object?>? reqdata,
        Dictionary<string, object?>? ctrl = null)
    {
        var ctx = utility.MakeContext(new Dictionary<string, object?>
        {
            ["opname"] = "create",
            ["ctrl"] = ctrl,
            ["match"] = match,
            ["data"] = data,
            ["reqdata"] = reqdata,
        }, entctx);
    
        return RunOp(ctx, () =>
        {
            if (ctx.Result != null)
            {
                if (ctx.Result.Resdata != null)
                {
                    data = Helpers.ToMapAny(
                        Voxgig.Struct.StructUtils.Clone(ctx.Result.Resdata))
                        ?? new Dictionary<string, object?>();
                }
            }
        });
    }

    // (update not defined by this API - base class throws UnsupportedOp)

    // (remove not defined by this API - base class throws UnsupportedOp)
}
